core/src/main/java/org/eclipse/hono/config/HonoConfigProperties.java

```diff
@@ -322,4 +322,23 @@ public HonoConfigProperties setWaitForDownstreamConnectionEnabled(final boolean
         this.waitForDownstreamConnection = waitForConnection;
         return this;
     }
+
+    public boolean handleUnconfiguredPort() {
+        boolean configuredPort = false;
+        if (getPort() == HonoConfigProperties.PORT_UNCONFIGURED) {
+            setPort(Constants.PORT_AMQP);
```

Comment:
I see - will pull it out and introduce a getPort(int) method like proposed then. Makes sense.